internal/consensus/replay.go

```diff
@@ -262,16 +262,17 @@ func (h *Handshaker) Handshake(ctx context.Context, appClient abciclient.Client)
 	// Only set the version if there is no existing state.
 	if h.initialState.LastBlockHeight == 0 {
 		h.initialState.Version.Consensus.App = res.AppVersion
+		h.initialState.ConsensusParams.Version.AppVersion = res.AppVersion
```

Comment:
just because I missed it the first time, this is the meaningful change in this patch, all the other lines are just renaming a variable (which I think is good.)

Reply:
yes, just one line change and the other lines are renaming!
I will add a test for it. Submit PR first just want to bring the discussion on the app_version issue.

Reply:
Can we instead make the assertion that they are the same? 

ConsensusParams has to be set (either from Genesis or the `InitChain` call)

Reply:
@cmwaters Do you mean we should check the consensusParam return from `InitChain` call with the AppVersion returns from the `Info` at the initial state? 

And we need to confirm the current setup of the kvstore app is reasonable or not (Which I don't know due to no SPEC describe it). The example app returns AppVersion 1 but doesn't return the consensusParam in `InitChain`, so the node using genesis's censensusparam cause the version mismatch.

Reply:
> Do you mean we should check the consensusParam return from InitChain call with the AppVersion returns from the Info at the initial state?

Yes I think so. 

> The example app returns AppVersion 1 but doesn't return the consensusParam in InitChain, so the node using genesis's censensusparam cause the version mismatch.

Then I guess we need to update that and the spec.

Reply:
updated by the feedback. the spec update will create an issue for it.